craft_cli/dispatcher.py

```diff
@@ -104,19 +116,19 @@ class BaseCommand:
     to the Dispatcher.
     """
 
+    name: str
+    help_msg: str
+    overview: str
     common = False
     hidden = False
-    name: Optional[str] = None
-    help_msg: Optional[str] = None
-    overview: Optional[str] = None
 
     def __init__(self, config: Optional[Dict[str, Any]]):
         self.config = config
 
         # validate attributes
         mandatory = ("name", "help_msg", "overview")
         for attr_name in mandatory:
-            if getattr(self, attr_name) is None:
+            if not hasattr(self, attr_name):
```

Comment:
Good catch! I updated this because the namedtuple change was breaking linting but obviously didn't think through all the edge cases.

I've updated it to still change the types and remove the default `None` values, but to also check for `None` values. This should turn it into a non-breaking change, making it fail in all the same cases. I added an explicit unit test for those failures as well.

Reply:
Ah even better ;) thanks!